File: Ejercicio1/calculadora.scala
println("---------- Calculadora ----------")
def suma(a:Int,b:Int):Int=a+b
def resta(a:Int,b:Int):Int=a-b
def mult(a:Int,b:Int):Int=a*b
def div(a:Int,b:Int):Int=a/b
def mod(a:Int,b:Int):Int=a%b

def calcu(x:Int, y:Int, operacion:String):Int={
    operacion match {
        case "suma"=>suma(x,y)
        case "resta"=>resta(x,y)
        case "mult"=>mult(x,y)
        case "div"=>div(x,y)
    }
}

var bool : Boolean = true;
while(bool == true){
    print("a:")
    val a = scala.io.StdIn.readInt()
    print("b:")
    val b = scala.io.StdIn.readInt()
    print("Operacion (suma, resta, mult o div):")
    val op = scala.io.StdIn.readLine()
    var res = 0
    if(op == "suma" || op == "resta" || op == "mult" || op == "div" ) {
        res = calcu(a,b,op)
        if(op == "div"){
            var m = mod(a,b) / b.toFloat
            var r = res.toFloat + m
            println(r)
        }
        else println(res)
    }
    else println("Operación no reconocida")
    println("Continuar? s/n")
    var co = scala.io.StdIn.readLine()
    if(co != "s") bool = false
}

File: Ejercicio1/calculadora.py
def suma(x,y):
    return x + y

def resta(x,y):
    return x - y

def mult(x,y):
    return x * y

def div(x,y):
    return x / y

def calcu(a,b,op):
    res = 0
    if op == 'suma':
        res = suma(a,b)
    if op == 'resta':
        res = resta(a,b)
    if op == 'mult':
        res = mult(a,b)
    if op == 'div':
        res = div(a,b)
    return res

bo = True
while(bo):
    print('--------- Calculadora ---------')
    a = int(input("a: "))
    b = int(input("b: "))
    op = input("Operación(suma, resta, mult, div): ")
    res = calcu(a,b,op)
    print(res)
    cont = input("Continuar? s/n \n")
    if(cont != "s"):
        bo = False

